com.unity.perception/Runtime/Randomization/Randomizers/SampleRandomizers/Randomizers/SunAngleRandomizer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using UnityEngine.Experimental.Perception.Randomization.Parameters;
+using UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers.Tags;
+using UnityEngine.Experimental.Perception.Randomization.Samplers;
+
+namespace UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers
+{
+    /// <summary>
+    /// Randomizes the rotation of directional lights tagged with a SunAngleRandomizerTag
+    /// </summary>
+    [Serializable]
+    [AddRandomizerMenu("Perception/Sun Angle Randomizer")]
+    public class SunAngleRandomizer : Randomizer
+    {
+        /// <summary>
+        /// The hour of the day (0 to 24)
+        /// </summary>
+        public FloatParameter hour = new FloatParameter { value = new UniformSampler(0, 24)};
```

Comment:
Since this is public, what happens if a user changes this to a float sampler between 0 and 1007. Do we have safeguards against that. I understand why you would want it public, because what if a user just wants to have a range between 5am and 2pm. But perhaps the public api should be two floats/ints for start time and end time that have to be between 0 and 24, and also make sure that start time < end time. Anyway, I may not completely understand what you have going on in the code, and this may not be an issue.

Reply:
There is a JIRA item open to address this limit setting behavior for samplers, but I think I'll hit it in a later release.